### ISSUE TITLE
NullReferenceException mapping dictionary to dictionary

### ISSUE THREAD

**Issue by kwaclaw** [agileobjects/AgileMapper#10]
This test case shows the issue. Same with code that is commented out.
```
        public class DictTest
        {
            public Dictionary<string, object> JobDataMap { get; set; }
        }

        public class DictTest2
        {
            public Dictionary<string, object> JobDataMap { get; set; }
        }

        [Fact]
        public void MapDicts() {
            var obj = new DictTest { JobDataMap = new Dictionary<string, object>() };
            var obj2 = Mapper.Map(obj).ToANew<DictTest2>();
            //var obj2 = new DictTest2 { JobDataMap = new Dictionary<string, object>() };
            //Mapper.Map(obj).Over(obj2);
        }
```

**Comment by SteveWilkes** [comment]
I've committed [a fix](https://github.com/agileobjects/AgileMapper/commit/a84b2ca0273412b73e650bb111444945d42e25cf) for this - I'll resolve the other open issues before doing another release.

**Comment by SteveWilkes** [comment]
Fixed in v0.10.1 - thanks for the bug report! :)

**Comment by kwaclaw** [comment]
It seems the fix is not entirely done. When I call the example above with actual data in the dictionary I get a MappingException. See this code:
```
        [Fact]
        public void MapDict() {
            var obj = new DictTest { JobDataMap = new Dictionary<string, object> { { "key1", new object() }, { "key2", new object() } } };
            var obj2 = Mapper.Map(obj).ToANew<DictTest2>();
        }
```

Some error details:
mapping InfrastructureTests.DictTest.JobDataMap -> InfrastructureTests.DictTest2.JobDataMap with rule set CreateNew.
-------- System.InvalidOperationException : Cannot jump to undefined label 'Return'.

**Comment by SteveWilkes** [comment]
Hmm - what data have you added to the dictionary? The following test passes for me:

```C#
[Fact]
public void ShouldMapADictionaryMemberToANewDictionaryMember()
{
    var source = new PublicField<Dictionary<string, object>>()
    {
        Value = new Dictionary<string, object>
        {
            ["Location"] = "I'm in a Dictionary!",
            ["Number"] = 123,
            ["Object"] = new CustomerViewModel { Name = "Mr Yo Yo" }
        }
    };

    var result = Mapper.Map(source).ToANew<PublicProperty<Dictionary<string, object>>>();

    result.Value.ShouldNotBeNull();
    result.Value["Location"].ShouldBe("I'm in a Dictionary!");
    result.Value["Number"].ShouldBe(123);
    result.Value["Object"].ShouldBeOfType<CustomerViewModel>();
    result.Value["Object"].ShouldNotBeSameAs(source.Value["Object"]);
    ((CustomerViewModel)result.Value["Object"]).Name.ShouldBe("Mr Yo Yo");
}
```

**Comment by kwaclaw** [comment]
I used the original code, but instead of an empty dictionary I used thisL
```
var obj = new DictTest { JobDataMap = new Dictionary<string, object> { { "key1", new object() }, { "key2", new object() } } };
```

**Comment by SteveWilkes** [comment]
Ahh, ok, I see. I don't think mapping objects of runtime-type 'object' is supported - I think it'd be a pretty simple fix so I'll definitely fix it, but I'll put it out with the next minor release within the next week.

Thanks again for the feedback :)

**Comment by SteveWilkes** [comment]
This should now be fixed as of [v0.10.2](https://www.nuget.org/packages/AgileObjects.AgileMapper).
